controllers/k8ssandra/cassandra_telemetry_reconciler.go

```diff
@@ -28,11 +28,11 @@ func (r *K8ssandraClusterReconciler) reconcileCassandraDCTelemetry(
 	cfg := telemetry.PrometheusResourcer{
 		MonitoringTargetNS:   actualDc.Namespace,
 		MonitoringTargetName: actualDc.Name,
-		ServiceMonitorName:   kc.Name + "-" + actualDc.Name + "-" + "cass-servicemonitor",
+		ServiceMonitorName:   kc.SanitizedName() + "-" + actualDc.Name + "-" + "cass-servicemonitor",
 		Logger:               logger,
-		CommonLabels:         mustLabels(kc.Name, kc.Namespace, actualDc.Name),
+		CommonLabels:         mustLabels(kc.SanitizedName(), kc.Namespace, actualDc.Name),
```

Comment:
```suggestion
		CommonLabels:         mustLabels(kc.Name, kc.Namespace, actualDc.Name),
```

Reply:
Agreed, labels should carry the `kc.Name` IMO.